src/unix/process.c

```diff
@@ -387,30 +387,21 @@ static void uv__spawn_init_posix_spawn_fncs(void) {
 
 
 static void uv__spawn_init_can_use_setsid(void) {
-  static const int MACOS_CATALINA_VERSION_MAJOR = 19;
-  char version_str[256];
-  char* version_major_str;
-  size_t version_str_size = 256;
-  int r;
-  int version_major;
-
-  /* Get a version string */
-  r = sysctlbyname("kern.osrelease", version_str, &version_str_size, NULL, 0);
-  if (r != 0)
+  int which[] = {CTL_KERN, KERN_OSRELEASE};
+  unsigned major;
+  unsigned minor;
+  unsigned patch;
+  char buf[256];
+  size_t len;
+
+  len = sizeof(buf);
+  if (sysctl(which, ARRAY_SIZE(which), buf, &len, NULL, 0))
     return;
 
-  /* Try to get the major version number. If not found
-   * fall back to the fork/exec flow */
-  version_major_str = strtok(version_str, ".");
-  if (version_major_str == NULL)
+  if (3 != sscanf(buf, "%u.%u.%u", &major, &minor, &patch))
```

Comment:
Looks like a good cleanup. We could also have used `atoi_l` without the `strtok` (since it stops when it hits a character it does not recognize), but this seems more complete. We can also avoid the dependence on the current locale (though it does not change much for %u) with the xlocale version:

```suggestion
  /* NULL specifies to use LC_C_LOCALE */
  if (3 != sscanf_l(buf, NULL, "%u.%u.%u", &major, &minor, &patch))
```